### PR TITLE
The Trusty Bucket - Introduction and Basics

### PR DESCRIPTION
**Pull request by brianknight10**
Hello!

Please review my introductory post to the AWS S3 service. This one is quite basic, covering what is S3, buckets, and objects. The aim is to layer on the complexity bit by bit with each post, but this one just introduces the series and level-sets on the basics.

Appreciate any and all feedback! Thanks for reviewing! :smile:

### CODE REVIEW

File: posts/the-trusty-bucket-introduction-and-basics.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+In the end, I hope you will see how the trusty and ubiquitous S3 bucket, while initially simple, gives you plenty of possibilities - and things to think about! - as you add them to your solution.
+
+# What is S3?
+
+S3 was the very first service launched as part of Amazon Web Services in March 2006. S3 stands for _Simple Storage Service_, and at first blush, it is just that - a simple storage service. You can think of it as a "giant disk drive in the sky." S3 provides storage for objects and files in the cloud and makes interacting with them accessible through a web interface. At a minimum, if you have an internet connection, you can access the stuff you have stored in S3 from anywhere in the world.
+
+## Buckets and Objects
+
+_Buckets_ are storage containers for your _objects_. A bucket is the highest level of organization within S3 and is a convenient boundary for billing aggregation and access control. You can choose a single [AWS Region(https://aws.amazon.com/about-aws/global-infrastructure/regions_az/)] for any bucket you create. Within that region, S3 will replicate your objects across multiple datacenters to ensure durability and reliability.
```
Comment (matthewreed26) [inline posts/the-trusty-bucket-introduction-and-basics.md:26]:
```suggestion
_Buckets_ are storage containers for your _objects_. A bucket is the highest level of organization within S3 and is a convenient boundary for billing aggregation and access control. You can choose a single [AWS Region](https://aws.amazon.com/about-aws/global-infrastructure/regions_az/) for any bucket you create. Within that region, S3 will replicate your objects across multiple datacenters to ensure durability and reliability.
```

File: posts/the-trusty-bucket-introduction-and-basics.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    Bucket=BUCKET_NAME,
+    CreateBucketConfiguration={
+        'LocationConstraint':BUCKET_REGION
+    }
+)
+```
+
+![A Bucket is Created](https://raw.githubusercontent.com/brianknight10/blog-usa/master/images/2021/02/trusty-bucket-1-2.png)
+
+## Checking out the bucket
```
Comment (matthewreed26) [inline posts/the-trusty-bucket-introduction-and-basics.md:63]:
```suggestion
## Checking out the bucket settings
```

File: posts/the-trusty-bucket-introduction-and-basics.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+# List all of the subresources for the bucket
+for resource in bucket.get_available_subresources():
+    print(resource)
+```
+
+![Bucket Sub-resources](https://raw.githubusercontent.com/brianknight10/blog-usa/master/images/2021/02/trusty-bucket-1-3.png)
+
+## Uploading our data
+
+An empty bucket is not very useful. We need to put some data in it! I always tell people that you could put a ham sandwich in S3, so I have photos of three delicious sandwiches that we can upload.
```
Comment (matthewreed26) [inline posts/the-trusty-bucket-introduction-and-basics.md:87]: Love this haha

File: posts/the-trusty-bucket-introduction-and-basics.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+)
+
+# Delete the bucket
+bucket.delete()
+```
+
+# Wrapping Up
+
+So far, so simple? I hope you enjoyed this quick summary of S3 and some of the basics of working with AWS's storage solution. Each subsequent post in this series will introduce another layer of functionality in S3. Please come back for the next post on sharing your S3 data with the world.
+
```
Comment (matthewreed26) [inline posts/the-trusty-bucket-introduction-and-basics.md:171]: Perhaps a few sources would be helpful to list at the end?

**Review by matthewreed26 — CHANGES_REQUESTED** [review]
Simple to follow. Very quick read. Good example. Just a few things I saw

**Comment by brianknight10** [comment]
Thanks @matthewreed26 - great suggestions!

**Review by matthewreed26 — APPROVED** [review]
LGTM

**Review by bescott1 — APPROVED** [review]

**Pull request merged**